src/symbol.js

```diff
@@ -5,17 +5,100 @@ import { fromSJSONDictionary, toSJSON } from 'sketchapp-json-plugin';
 import StyleSheet from './stylesheet';
 import { generateID } from './jsonUtils/models';
 import ViewStylePropTypes from './components/ViewStylePropTypes';
-import type { SketchContext } from './types';
 import buildTree from './buildTree';
 import flexToSketchJSON from './flexToSketchJSON';
+import { replaceAllLayersWithLayers } from './render';
 
 let id = 0;
 const nextId = () => ++id; // eslint-disable-line
 
 const displayName = (Component: React$Component): string =>
   Component.displayName || Component.name || `UnknownSymbol${nextId()}`;
 
-const mastersNameRegistry = {};
+let mastersNameRegistry = null;
+let existingSymbols = null;
+const layers = {};
+
+const msListToArray = (pageList) => {
+  const out = [];
+  // eslint-disable-next-line
+  for (let i = 0; i < pageList.length; i++) {
+    out.push(pageList[i]);
+  }
+  return out;
+};
+
+export const getExistingSymbols = () => {
+  const globalContext = context; // eslint-disable-line
+  const pages = globalContext.document.pages();
+  const array = msListToArray(pages);
+  if (existingSymbols === null) {
+    let symbolsPage = array.find(p => String(p.name()) === 'Symbols');
+    if (!symbolsPage) {
+      symbolsPage = globalContext.document.addBlankPage();
+      symbolsPage.setName('Symbols');
+    }
+
+    existingSymbols = msListToArray(symbolsPage.layers()).map((x) => {
+      const symbolJson = JSON.parse(toSJSON(x));
+      layers[symbolJson.symbolID] = x;
+      return symbolJson;
+    });
+
+    mastersNameRegistry = {};
+    existingSymbols.forEach((symbolMaster) => {
+      if (symbolMaster._class !== 'symbolMaster') return;
+      if (symbolMaster.name in mastersNameRegistry) return;
+      mastersNameRegistry[symbolMaster.name] = symbolMaster;
+    });
+  }
+  return existingSymbols;
+};
+
+export const getSymbolId = (masterName: string): string => {
+  let symbolId = generateID();
+
+  existingSymbols.forEach((symbolMaster) => {
+    if (symbolMaster.name === masterName) {
+      symbolId = symbolMaster.symbolID;
+    }
+  });
+  return symbolId;
+};
+
+const injectSymbols = () => {
+  const globalContext = context; // eslint-disable-line
+  const pages = globalContext.document.pages();
+  const array = msListToArray(pages);
+
+  let symbolsPage = array.find(p => String(p.name()) === 'Symbols');
```

Comment:
FYI, there's a `symbolsPageOrCreateIfNecessary()` method in MSDocumentData that does exactly what you want (and supports a Symbols page with a name other than 'Symbols' : )